source/docs/software/wpilib-tools/shuffleboard/custom-widgets/creating-a-new-widget.rst

```diff
@@ -0,0 +1,199 @@
+
+.. code-block:: xml
+
+   <?import javafx.scene.layout.*?>
+   <GridPane xmlns:fx="http://javafx.com/fxml/1" fx:controller="/path/to/widget/class" fx:id="root">
+
+      <VBox>
+         <Slider fx:id = "xSlider"/>
+         <Slider fx:id = "ySlider"/>
+      </VBox>
+
+   </GridPane>
+
+Now that we have finished creating our FXML file, we can now create a widget class. The widget class should include a ``@Description`` annotation that states the supported data types of the widget and the name of the widget. Although not required, not adding the ``@Description`` annotation must implement the ``get()`` method in the ``AbstractWidgetType`` class.
+
+It also must include a ``@ParamatrizedController`` annotation that points to the FXML file containing the layout of the widget. If the class that only supports one data source it must extend the ``SimpleAnnotatedWidget`` class. If the class supports multiple data sources, it must extend the ``ComplexAnnotatedWidget`` class.
+
+
+.. code-block:: java
+
+   import edu.wpi.first.shuffleboard.api.widget.SimpleAnnotatedWidget;
+   import edu.wpi.first.shuffleboard.api.widget.Description;
+   import edu.wpi.first.shuffleboard.api.widget.ParametrizedController;
+
+   /*If the FXML file and Java file are in the same package, that is the java file is in src/main/java and the
+    * FXML file is under src/main/resources or your code equivalent package, the relative path will work
+    * However, if they are in different packages, an absoulte path will be required.
+   */
+
+   @Description(name = "MyPoint2D", dataTypes = MyPoint2D.class)
+   @ParamatrizedController("Point2DWidget.fxml")
+   public final class Point2DWidget extends SimpleAnnotatedWidget<MyPoint2D> {
+
+   }
+
+If you are not using a custom data type, you can reference any Java data type (ie. double.class) or if the widget does not need data binding you can pass ``NoneType.class``.
+
+Now that we have created our class we can create fields for the widgets we declared in our FXML file using the ``@FXML`` annotation. For our two sliders, an example would be:
+
+.. code-block:: java
+
+   import edu.wpi.first.shuffleboard.api.widget.SimpleAnnotatedWidget;
+   import javafx.fxml.FXML;
+   import edu.wpi.first.shuffleboard.api.widget.Description;
+   import edu.wpi.first.shuffleboard.api.widget.ParametrizedController;
+
+   @Description(name = "MyPoint2D", dataTypes = MyPoint2D.class)
+   @ParametrizedController("Point2DWidget.fxml")
+   public final class Point2DWidget extends SimpleAnnotatedWidget<MyPoint2D> {
+      @FXML
+      private Pane root;
+
+      @FXML
+      private Slider xSlider;
+
+      @FXML
+      private Slider ySlider;
+   }
+
+In order to display our pane on our custom widget we need to override the ``getView()`` method and return our ``StackedPane``.
```

Comment:
```suggestion
In order to display our pane on our custom widget we need to override the ``getView()`` method and return our ``StackPane``.
```

Reply:
Also, the docs here say `StackPane` but the FXML uses `GridPane`. Pick one and stick to it